Refactored DamageCause and EntityDamageEvent

Mattier opened the pull request:
Needed to properly implement EntityDamageByBlock events in Bridge.

Signed-off-by: Charlotte Mattier halfbloodpokemon@gmail.com

Afforess commented on src/main/java/org/spout/vanilla/event/cause/DamageCause.java:
```diff
@@ -0,0 +1,156 @@
+	 * Sets the type of the damage cause.
+	 * @return type
+	 */
+	public void setType(DamageType type) {
```
Do we really want setters?

Afforess commented on src/main/java/org/spout/vanilla/event/cause/DamageCause.java:
```diff
@@ -0,0 +1,156 @@
+	 * Sets the source of this damage cause.
+	 * @param source
+	 */
+	public void setSource(Object source) {
```
Do we really want setters?

Mattier commented:
They aren't necessary. Vanilla EntityDamageEvent already had setters for the damager and damage type, so I carried them over without thinking too much about it. I can remove those from the damage source and damage event.

Commit:
Refactored DamageCause and EntityDamageEvent, so that any object can be
a damage source, as opposed to only an entity. Needed to implement
EntityDamageByBlock events in Bridge.

Signed-off-by: Charlotte Mattier <halfbloodpokemon@gmail.com>
